test/unix/main.ml

```diff
@@ -332,6 +332,21 @@ module Close = struct
     Alcotest.check_raises "flush after double close with raises Closed"
       I.Closed (fun () -> Index.flush rw)
 
+  let restart_twice () =
+    let reuse_name = Context.fresh_name "empty_index" in
+    let rw = Index.v ~fresh:true ~readonly:false ~log_size:4 reuse_name in
+    let k1, v1 = (Key.v (), Value.v ()) in
+    Index.replace rw k1 v1;
+    Index.close rw;
+    let rw = Index.v ~fresh:true ~readonly:false ~log_size:4 reuse_name in
+    Alcotest.check_raises "Index restarts fresh cannot read data." Not_found
+      (fun () -> ignore (Index.find rw k1));
```

Comment:
It's paranoid, but you could define an `ignore_value` utility function with a constrained type to avoid accidentally ignoring a partially-applied function if the `find` type ever changes:
```ocaml
let ignore_val (_ : value) = () in
```

Reply:
Nice trick! Thanks!